airbyte-commons/src/main/java/io/airbyte/commons/io/LineGobbler.java

```diff
@@ -68,8 +86,10 @@ public void voidCall() {
       while ((line = is.readLine()) != null) {
         consumer.accept(line);
       }
+    } catch (IOException i) {
```

Comment:
This can happen if the Gobbler tries to read the error stream after the process has been closed.

The first thing I tried was adding an atomic boolean `running`, changing the while loop to be `while(running.get() && (line = is.readLine()) != null)` and a `close` method to be called before the process is closed.

However this was still happening since the boolean evaluation isn't atomic i.e. `running` is set to false after we evaluate it and the input stream is closed. We could use a lock here, but that felt like too much complexity for little gain. What do you think?